docs/using_docker.md

```diff
@@ -0,0 +1,160 @@
+```
+
+For binding a local folder to your *btcd* container please read the [Docker documentation](https://docs.docker.com/). The preferred way is to use a Docker managed volume.
+
+## Known error messages when starting the btcd container
+
+We pass all needed arguments to *btcd* as command line parameters in our *docker-compose.yml* file. It doesn't make sense to create a *btcd.conf* file. This would make things too complicated. Anyhow *btcd* will complain with following log messages when starting. These messages can be ignored:
+
+```bash
+Error creating a default config file: open /sample-btcd.conf: no such file or directory
+...
+[WRN] BTCD: open /root/.btcd/btcd.conf: no such file or directory
+```
+
+## Examples
+
+### Preamble
+
+All following examples uses some defaults:
+
+* container_name: btcd
+  Name of the docker container that is be shown by e.g. ```docker ps -a```
+
+* hostname: btcd **(very important to set a fixed name before first start)**
+  The internal hostname in the docker container. By default docker is recreating the hostname every time you change the *docker-compose.yml* file. The default hostnames look like *ef00548d4fa5*. This is a problem when using the *btcd* RPC port. The RPC port is using a certificate to validate the hostname. If the hostname changes you need to recreate the certificate. To avoid this, you should set a fixed hostname before the first start. This ensures, that the docker volume is created with a certificate with this hostname.
+
+* restart: unless-stopped
+  Starts the *btcd* container when Docker starts, except that when the container is stopped (manually or otherwise), it is not restarted even after Docker restarts.
+
+To use the following examples create an empty directory. In this directory create a file named *docker-compose.yml*, copy and paste the example into the *docker-compose.yml* file and run it.
+
+```bash
+mkdir ~/btcd-docker
+cd ~/btcd-docker
+touch docker-compose.yaml
+nano docker-compose.yaml (use your favourite editor to edit the compose file)
+docker-compose up (creates and starts a new btcd container)
+```
+
+With the following commands you can control *docker-compose*:
+
+```docker-compose up -d``` (creates and starts the container in background)
+
+```docker-compose down``` (stops and delete the container. **The docker volume btcd-data will not be deleted**)
+
+```docker-compose stop``` (stops the container)
+
+```docker-compose start``` (starts the container)
+
+```docker ps -a``` (list all running and stopped container)
+
+```docker volume ls``` (lists all docker volumes)
+
+```docker logs btcd``` (shows the log )
+
+```docker-compose help``` (brings up some helpful information)
+
+### Full node without RPC port
+
+Let's start with an easy example. If you just want to create a full node without the need of using the RPC port, you can use the following example. This example will launch *btcd* and exposes only the default p2p port 8333 to the outside world:
+
+```dockerfile
```

Comment:
nit *yaml* would make this more readable.

Reply:
fixed